### PR TITLE
Outtax sec commod

### PR DESCRIPTION
**Pull request by EliLazarus**
The fix was simple and works. I think we should merge it.
I think the ongoing mismatch in results in the model tests I'm getting are unrelated/peripherally related to this issue, so better for a separate PR.

### CODE REVIEW

**Comment by EliLazarus** [comment]
This is ready for Prime Time.
There are 3 (main) fixes:
- I pass the production to `get_jump_expression_for_commodity_producer_price` and then delete the loop through productions from that function
- I updated to `get_jump_expression_for_commodity_producer_price(m, jm, s, output.commodity))` in adding the compensated supply expression
- I included a "fix" for fixing consumer values. We now only calculate it as the updated sum of endowments and taxes when not fixing its value. This was needed to check against GAMS results, and is the GAMS behaviour.

Then I added tests, and updated all tests to include setting the Consumer value before fixing because that is now required. That matches the GAMS version. I don't think there's a way around that because the user can fix at the original benchmark, or at an updated value for a counter-factual. We (and GAMS) default to the benchmark value if fixed without setting a value.

File: src/build/build_helpers.jl [hunk 1]
```diff
@@ -179,11 +178,11 @@ function get_tax_revenue_for_consumer(jm, m, cr::ConsumerRef)
             for tax in output.taxes
                 if cr.subindex === nothing
                     if get_full(tax.agent) == get_full(cr)    
-                        push!(taxes, :($(tax.rate) * $(output.quantity) * $(output.commodity) * $(pf.sector)))
+                        push!(taxes, :($(tax.rate) * $jm[get_comp_supply_name($output)] * $(output.commodity) * $(pf.sector)))
```
Comment (davidanthoff) [inline src/build/build_helpers.jl:181]: Are you sure this works? I think what would happen here is that the value of `jm` gets interpolated into the expression, and then the expression continues with `[get_comp_supply_name(`. But that is presumably not what we want? I.e. we would want `get_comp_supply_name` to evaluate while we build the expression, right? I would think it needs to be `$(jm[get_comp_supply_name(output)])`. Same for the next diff below.

**Review by davidanthoff — CHANGES_REQUESTED** [review]
One small change, I think.

File: src/build/build_helpers.jl (hunk 1 above)
Comment (EliLazarus) [reply src/build/build_helpers.jl]: It _seems_ to work either way, but I updated as above.

Comment (davidanthoff) [reply src/build/build_helpers.jl]: Yeah, it might work, but the new way should be more efficient, I think :)

**Pull request merged**